Blish HUD/GameServices/Contexts/CdnInfoContext.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Threading.Tasks;
+using Flurl.Http;
+
+namespace Blish_HUD.Contexts {
+
+    public class CdnInfoContext : Context {
+
+        private static readonly Logger Logger = Logger.GetLogger(typeof(CdnInfoContext));
+        
+        public struct CdnInfo {
+
+            private readonly int _buildId;
+            private readonly int _exeFileId;
+            private readonly int _exeFileSize;
+            private readonly int _manifestFileId;
+            private readonly int _manifestFileSize;
+
+            public int BuildId          => _buildId;
+            public int ExeFileId        => _exeFileId;
+            public int ExeFileSize      => _exeFileSize;
+            public int ManifestFileId   => _manifestFileId;
+            public int ManifestFileSize => _manifestFileSize;
+
+            public CdnInfo(string rawCdnResponse) : this() {
+                string[] cdnVars = rawCdnResponse.Split(' ');
+
+                bool parsedSuccessfully = true;
+
+                if (cdnVars.Length == 5) {
+                    parsedSuccessfully &= int.TryParse(cdnVars[0], out _buildId);
+                    parsedSuccessfully &= int.TryParse(cdnVars[1], out _exeFileId);
+                    parsedSuccessfully &= int.TryParse(cdnVars[2], out _exeFileSize);
+                    parsedSuccessfully &= int.TryParse(cdnVars[3], out _manifestFileId);
+                    parsedSuccessfully &= int.TryParse(cdnVars[4], out _manifestFileSize);
+                }
+
+                if (!parsedSuccessfully) {
```

Comment:
minor bug: `parsedSuccessfully` can return the wrong result

```cs
bool parsedSuccessfully = false;

if (cdnVars.Length == 5) {
    parsedSuccessfully = true;
    ...
}
if (!parsedSuccessfully) {

Reply:
I would like to advise against doing that. the compiler should be able to optimize both solutions, but a normal developer would definitely oversee that you didnt write `==`